### PR TITLE
SceneQueryRunner: Manual control over query execution

### PR DESCRIPTION
**Pull request by torkelo**
This PR adds a new prop `runQueriesMode` to QueryRunnerState.
The default value is "auto": which should not change query runner behavior.
Setting runQueriesMode to "manual", will:
* Not fire queries on activate
* Not fire queries on time range update
* Not fire queries on variable updates (including adhoc/group by variables)

TL;DR: When setting runQueriesMode is set to manual, queries must be executed via `runQueries`.

The only exception is when `maxDataPointsFromWidth` is also set, then the query will be run when the container width changes.

If folks want more control we could follow up this PR with more options to the `runQueriesMode`, e.g. only automatically firing queries when variables are updated, or on activate, etc, but for now the new mode provides an option for users that need more control over when queries are run.

<details>
  <summary>📦 Published PR as canary version: <code>5.9.0--canary.334.10407764115.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/scenes-react@5.9.0--canary.334.10407764115.0
  npm install @grafana/scenes@5.9.0--canary.334.10407764115.0
  # or 
  yarn add @grafana/scenes-react@5.9.0--canary.334.10407764115.0
  yarn add @grafana/scenes@5.9.0--canary.334.10407764115.0
  ```
</details>

### CODE REVIEW

**Comment by andyfleming-shopify** [comment]
This would be a useful feature to have for the scene query runner 👍 .

**Comment by gtk-grafana** [comment]
@torkelo I can't assign the review to you as you created the original PR, but if you can take a look I'd appreciate it!

File: packages/scenes/src/querying/SceneQueryRunner.ts [hunk 1]
```diff
@@ -260,12 +271,14 @@ export class SceneQueryRunner extends SceneObjectBase<QueryRunnerState> implemen
       return;
     }
 
-    if (variable instanceof AdHocFiltersVariable && this._isRelevantAutoVariable(variable)) {
-      this.runQueries();
-    }
+    if(this.isQueryModeAuto()){
+      if (variable instanceof AdHocFiltersVariable && this._isRelevantAutoVariable(variable)) {
```
Comment (torkelo) [inline packages/scenes/src/querying/SceneQueryRunner.ts:275]: would add this to the top of the function (the early return statement), !this.isQueryModeAuto() , to reduce the need for nested ifs

Comment (gtk-grafana) [reply packages/scenes/src/querying/SceneQueryRunner.ts]: Fixed!

**Comment by torkelo** [comment]
Looks good , now just needs a review approval

**Review by matyax — APPROVED** [review]

**Pull request merged**